src/main/java/appeng/fluids/parts/PartFluidAnnihilationPlane.java

```diff
@@ -0,0 +1,322 @@
+	}
+
+	private void refresh()
+	{
+		try
+		{
+			this.getProxy().getTick().alertDevice( this.getProxy().getNode() );
+		}
+		catch( final GridAccessException e )
+		{
+			// :P
+		}
+	}
+
+	@Override
+	@MENetworkEventSubscribe
+	public void chanRender( final MENetworkChannelsChanged c )
+	{
+		this.refresh();
+		this.getHost().markForUpdate();
+	}
+
+	@Override
+	@MENetworkEventSubscribe
+	public void powerRender( final MENetworkPowerStatusChange c )
+	{
+		this.refresh();
+		this.getHost().markForUpdate();
+	}
+
+	private TickRateModulation pickupFluid()
+	{
+		if( !this.getProxy().isActive() )
+		{
+			return TickRateModulation.SLEEP;
+		}
+
+		final TileEntity te = this.getTile();
+		final World w = te.getWorld();
+		final BlockPos pos = te.getPos().offset( this.getSide().getFacing() );
+		final IBlockState state = w.getBlockState( pos );
+		final Block block = state.getBlock();
+
+		if( block instanceof IFluidBlock || block instanceof BlockLiquid )
+		{
+			final IFluidHandler fh = FluidUtil.getFluidHandler( w, pos, null );
+			final IAEFluidStack blockFluid = AEFluidStack.fromFluidStack( fh.drain( Integer.MAX_VALUE, false ) );
+
+			if( blockFluid != null )
+			{
+				if( this.storeFluid( blockFluid, false ) )
+				{
+					this.storeFluid( AEFluidStack.fromFluidStack( fh.drain( Integer.MAX_VALUE, true ) ), true );
+
+					AppEng.proxy.sendToAllNearExcept( null, pos.getX(), pos.getY(), pos.getZ(), 64, w,
+							new PacketTransitionEffect( pos.getX(), pos.getY(), pos.getZ(), this.getSide(), true ) );
+
+					return TickRateModulation.URGENT;
+				}
+				return TickRateModulation.IDLE;
+			}
+		}
+
+		// nothing to do here :)
+		return TickRateModulation.SLEEP;
+	}
+
+	@Override
+	public TickingRequest getTickingRequest( final IGridNode node )
+	{
+		return new TickingRequest( TickRates.AnnihilationPlane.getMin(), TickRates.AnnihilationPlane.getMax(), false, true );
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( final IGridNode node, final int ticksSinceLastCall )
+	{
+		return this.pickupFluid();
+	}
+
+	private boolean storeFluid( IAEFluidStack stack, boolean modulate )
+	{
+		try
+		{
+			final IStorageGrid storage = this.getProxy().getStorage();
+			final IMEInventory<IAEFluidStack> inv = storage.getInventory( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+
+			if( modulate )
+			{
+				final IEnergyGrid energy = this.getProxy().getEnergy();
+				return Platform.poweredInsert( energy, inv, stack, this.mySrc ) == null;
+			}
+			else
+			{
+				final float requiredPower = stack.getStackSize() / Math.min( 1, stack.getChannel().transferFactor() );
+				final IEnergyGrid energy = this.getProxy().getEnergy();
+
+				if( energy.extractAEPower( requiredPower, Actionable.SIMULATE, PowerMultiplier.CONFIG ) < requiredPower )
+				{
+					return false;
+				}
+				return inv.injectItems( stack, Actionable.SIMULATE, this.mySrc ) == null;
+			}
+		}
+		catch( final GridAccessException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)